src/injectIntoComponent.js

```diff
@@ -0,0 +1,46 @@
+import { PropTypes } from 'react';
+import Injector from './Injector';
+/**
+ * Creates an extended component that will use injector-js DI to inject into the context.
+ * @param  {React.Component} Component The Component To Extend
+ * @return {React.Component}           Extended Component
+ */
+export default function injectIntoComponent(Component) {
```

Comment:
You might also consider making this a higher order component.  It's a fairly clean way to do something like this.
https://gist.github.com/sebmarkbage/ef0bf1f338a7182b6775

```
import { Component } from "React";

export var injectInto = ComposedComponent => class extends Component {

  static get propTypes() {
    return {
      injector: PropTypes.instanceof(Injector).isRequired
    }
  }

  getChildContext() {
    ...(
      injectables.reduce((context, [key]) => {
        return {
          ...context,
          [key]: this.props.injector.getInstance(key)
        };
    }, {})
  }

  render() {
    return <ComposedComponent {...this.props} />;
  }
};
```